### PR TITLE
feat: cypress integration tests

### PR DESCRIPTION
**Pull request by UnicornChance**
## Description

Adding cypress integration testing for UDS Core + UDS Identity Config. This tests user authentication and registration processes that can be affected by changing Identity config images. 

At this time it's utilizing a test realm.json that disables email verification, MFA, and OCSP checking. Obviously this creates a disparity between realm files, so[ this issue](https://github.com/defenseunicorns/uds-identity-config/issues/41) has been created for hopefully building out more robust integration tests that can also test the email verification and MFA. 

Has been re-based against main, so merge conflicts should be resolved.

Tested this in private CI and its work as well as locally. Docs have been included to explain the uds task as well as the different currently implemented tests and how to update the certs that are used by the tests.

To run the tests:
`uds run uds-core-integration-tests`

## Related Issue
Relates to #27

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

**Review by rjferguson21 — DISMISSED** [review]

File: docs/INTEGRATION.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  - Deploy zarf package that was created earlier
+  - Run cypress tests against deployed cluster
+
+## Updating Cypress Certs
+
+Cypress testing requires that a ca.cer be created and put into an authorized_certs.zip, done by using the `regenerate-test-pki` uds task, which is then utilized by the [Dockerfile](../src/Dockerfile). Once a docker image has been created another command is used for pulling that cacert, uds task `cacert`, from the image using it's value to configure uds-core's gateways, `uds-core-gateway-cacert` uds task . Eventually cypress will require a [pfx cert](../src/test/cypress/cypress.config.ts) for its CAC testing. 
+
+Our cypress testing utilizes [static certs](../src/test/cypress/certs/) that are created and saved to limit the need for constantly rebuilding and importing those certs.
+
+Follow these steps to update the certs for cypresss:
```
Comment (mjnagel) [inline docs/INTEGRATION.md:36]:
```suggestion
Follow these steps to update the certs for cypress:
```

File: zarf.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 kind: ZarfPackageConfig
 metadata:
   name: keycloak-identity-config
-  version: "0.0.1"
+  version: dev
```
Comment (mjnagel) [inline zarf.yaml:4]: @blancharda just want to double check with you - is this change okay? I think you were using this zarf package for an environment, want to make sure we don't mess with the version if its going to cause pain.

**Review by mjnagel — COMMENTED** [review]
Overall this looks great - really appreciate the work that went into it!

File: zarf.yaml (hunk 2 above)
Comment (blancharda) [reply zarf.yaml:4]: We didn't actually end up using this, so no immediate impact here 👍

**Review by mjnagel — APPROVED** [review]

**Pull request merged**